src/libraries/Hooks.sol

```diff
@@ -73,6 +73,26 @@ library Hooks {
     /// @notice The hook's delta changed the swap from exactIn to exactOut or vice versa
     error HookDeltaExceedsSwapAmount();
 
+    /// @notice Function to calculate flags from permissions
+    /// @param permissions The permissions to convert to flags
+    /// @return out The flags that correspond to the permissions
+    function flags(Hooks.Permissions memory permissions) internal pure returns (uint160 out) {
+        if (permissions.beforeInitialize) out |= Hooks.BEFORE_INITIALIZE_FLAG;
+        if (permissions.afterInitialize) out |= Hooks.AFTER_INITIALIZE_FLAG;
```

Comment:
```suggestion
        assembly {
            // out = permissions.beforeInitialize << BEFORE_INITIALIZE_BIT
            out := shl(BEFORE_INITIALIZE_BIT, mload(permissions))
            // out |= permissions.afterInitialize << AFTER_INITIALIZE_BIT
            out := or(out, shl(AFTER_INITIALIZE_BIT, mload(add(permissions, 0x20))))
        }
```
nit: declare
```solidity
    uint8 internal constant BEFORE_INITIALIZE_BIT = 13;
    uint160 internal constant BEFORE_INITIALIZE_FLAG = 1 << BEFORE_INITIALIZE_BIT;
    uint8 internal constant AFTER_INITIALIZE_BIT = 12;
    uint160 internal constant AFTER_INITIALIZE_FLAG = 1 << AFTER_INITIALIZE_BIT;
```
and use assembly before Solidity allows casting `bool` to `uint`.